### DISCUSSION TITLE
Documentation on the bindings

### DISCUSSION THREAD

**Discussion by PlatinumLucario** [gircore/gir.core#777] [Q&A] [answered]
Since this is my first time trying out and experimenting around with Gir.Core, there was some stuff that I came across that I had no idea on what it does. For example, the difference between ``Gtk.Internal.Box`` and ``Gtk.Box``.

I'm aware of all the C documentation on https://docs.gtk.org/, and I can use that as a reference for figuring out each of the different functions, but some of them, like ``Gtk.Internal`` don't have a page about it on the official GTK docs page.

So I'm wondering, is there any plans on creating documentation on all the bindings in Gir.Core?

**Comment by badcel** [comment, marked as answer]
Generating some API documentation is planned for 0.4.0 (see #411). But this will probably pretty rudimentary.

In general the Bindings are created in a 2 Layer approach. E.g. users of the bindings should use the Namespace `Gtk` and not `Gtk.Internal` as the internal namespace only contains internal code which is not as easy to use from an enduser perspective. The classes living in the `Gtk` namespace will call into `Gtk.Internal` to make things work.

As the Bindings are not complete yet and often users have special needs all those internal bindings are publicly available and only moved into the `internal` namespace to allow users to dig into the low level stuff if needed.

As an example `Gtk.Box` is the `Box` class as you would expect. `Gtk.Internal.Box` is just a helper class which provides all the methods on a `Box` class but it is requiring pointers as parameters which is not very convenient. If you use the `Gtk.Box` class it will itself call into `Gtk.Internal.Box.xxx` and do it's magic.
